lib/astutils.cpp

```diff
@@ -133,9 +133,85 @@ const Token * astIsVariableComparison(const Token *tok, const std::string &comp,
     return ret;
 }
 
+static const Token * getVariableInitExpression(const Variable * var)
+{
+    if(!var || !var->declEndToken())
+        return nullptr;
+    if(Token::Match(var->declEndToken(), "; %varid% =", var->declarationId()))
+        return var->declEndToken()->tokAt(2)->astOperand2();
+    return var->declEndToken()->astOperand2();
+}
+
+static bool isInLoopCondition(const Token * tok) {
+    return Token::Match(tok->astTop()->previous(), "for|while (");
+}
+
+
+/// This takes a token that refers to a variable and it will return the token
+/// to the expression that the variable is assigned to. If its not valid to
+/// make such substitution then it will return the original token.
+static const Token * followVariableExpression(const Token * tok, bool cpp)
+{
+    if(!tok)
+        return tok;
+    // Skip array access
+    if(Token::Match(tok, "%var% ["))
+        return tok;
+    // Skip pointer indirection
+    if(Token::Match(tok->previous(), "* %var%") && !tok->previous()->astOperand2())
```

Comment:
I added `Token::isUnaryOp()`

Reply:
I dont see it on master.

Reply:
sorry I forgot to push it.. it's pushed now

Reply:
sorry I missed pushing it. it has been pushed now.